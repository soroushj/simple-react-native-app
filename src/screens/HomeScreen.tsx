import React from 'react';
import { View, Text, Button } from 'react-native';
import { NavigationScreenOptions } from 'react-navigation';
import { IHomeScreenProps } from './HomeScreen.interfaces';
import styles from './styles';

export default class HomeScreen extends React.Component<IHomeScreenProps, {}> {
  static navigationOptions: NavigationScreenOptions = {
    title: 'Home',
  };

  onCounterPress = () => {
    this.props.navigation.navigate('Counter');
  };

  render() {
    return (
      <View style={styles.screenContainer}>
        <View style={styles.elementContainer}>
          <Text>Welcome!</Text>
        </View>
        <View style={styles.elementContainer}>
          <Button title=" Counter " onPress={this.onCounterPress} />
        </View>
      </View>
    );
  }
}
