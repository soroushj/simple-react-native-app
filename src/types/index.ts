export interface IRootState {
  counter: number;
}
