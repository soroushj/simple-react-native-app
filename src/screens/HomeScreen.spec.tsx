import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HomeScreen from './HomeScreen';
import { IHomeScreenProps } from './HomeScreen.types';

const setup = () => {
  const props: IHomeScreenProps = {
    navigation: { navigate: jest.fn() } as any,
  };
  const wrapper = shallow(<HomeScreen {...props} />);
  return { wrapper, props };
};

test('HomeScreen snapshot', () => {
  const { wrapper } = setup();
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('HomeScreen, counterButton press', () => {
  const { wrapper, props } = setup();
  const counterButton = wrapper.find('[testID="counterButton"]');
  const counterButtonProps = counterButton.props() as any;
  counterButtonProps.onPress();
  expect(props.navigation.navigate).toBeCalledWith('Counter');
});
