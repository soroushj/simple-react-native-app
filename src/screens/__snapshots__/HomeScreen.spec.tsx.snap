// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen snapshot 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Component
    style={
      Object {
        "margin": 10,
      }
    }
  >
    <Component>
      Welcome!
    </Component>
  </Component>
  <Component
    style={
      Object {
        "margin": 10,
      }
    }
  >
    <Button
      onPress={[Function]}
      testID="counterButton"
      title=" Counter "
    />
  </Component>
</Component>
`;
